src/sentry/models/avatars/base.py

```diff
@@ -103,21 +101,32 @@ def get_cached_photo(self, size):
                 cache.set(cache_key, photo)
         return photo
 
-    @classmethod
-    def file_class(cls):
-        from sentry.models import File
-
+    def file_class(self):
         return File
 
-    @classmethod
-    def file_fk(cls) -> str:
+    def file_fk(self) -> str:
+        """
+        Get the foreign key currently used by this record for blob storage.
+        Varies in ControlAvatarBase
+        """
+        return "file_id"
+
+    def file_write_fk(self) -> str:
+        """
+        Get the foreign key that should be used for writes.
+        Varies in ControlAvatarBase
+        """
         return "file_id"
 
     @classmethod
     def save_avatar(cls, relation, type, avatar=None, filename=None, color=None):
         if avatar:
-            with atomic_transaction(using=router.db_for_write(cls.file_class())):
-                photo = cls.file_class().objects.create(name=filename, type=cls.FILE_TYPE)
+            # Create an instance of the current class so we can
+            # access where new files should be stored.
+            dummy = cls()
```

Comment:
nit: is it fair to categorize this as a dummy class? Shouldn't this theoretically just return whatever the implementing class currently is?

Reply:
I went down this path because `save_avatar` is used for both region and control files. We save the file *before* the avatar record, and the avatar record doesn't always exist so I can't use the instance properties from that object.

Using a 'dummy' (not a great name) instance gives me a way to determine where the file write *should* go.